src/qtoolkit/io/sge.py

```diff
@@ -0,0 +1,526 @@
+from __future__ import annotations
+
+import re
+import xml.dom.minidom
+import xml.parsers.expat
+from datetime import timedelta
+
+from qtoolkit.core.data_objects import (
+    CancelResult,
+    CancelStatus,
+    ProcessPlacement,
+    QJob,
+    QJobInfo,
+    QResources,
+    QState,
+    QSubState,
+    SubmissionResult,
+    SubmissionStatus,
+)
+from qtoolkit.core.exceptions import OutputParsingError, UnsupportedResourcesError
+from qtoolkit.io.base import BaseSchedulerIO
+
+# 'http://www.loni.ucla.edu/twiki/bin/view/Infrastructure/GridComputing?skin=plain':
+# Jobs Status:
+#     'qw' - Queued and waiting,
+#     'w' - Job waiting,
+#     's' - Job suspended,
+#     't' - Job transferring and about to start,
+#     'r' - Job running,
+#     'h' - Job hold,
+#     'R' - Job restarted,
+#     'd' - Job has been marked for deletion,
+#     'Eqw' - An error occurred with the job.
+#
+# 'http://confluence.rcs.griffith.edu.au:8080/display/v20zCluster/
+# Sun+Grid+Engine+SGE+state+letter+symbol+codes+meanings':
+#
+# Category     State     SGE Letter Code
+# Pending:     pending     qw
+# Pending:     pending, user hold     qw
+# Pending:     pending, system hold     hqw
+# Pending:     pending, user and system hold     hqw
+# Pending:     pending, user hold, re-queue     hRwq
+# Pending:     pending, system hold, re-queue     hRwq
+# Pending:     pending, user and system hold, re-queue     hRwq
+# Pending:     pending, user hold     qw
+# Pending:     pending, user hold     qw
+# Running     running     r
+# Running     transferring     t
+# Running     running, re-submit     Rr
+# Running     transferring, re-submit     Rt
+# Suspended     job suspended     s, ts
+# Suspended     queue suspended     S, tS
+# Suspended     queue suspended by alarm     T, tT
+# Suspended     all suspended with re-submit     Rs, Rts, RS, RtS, RT, RtT
+# Error     all pending states with error     Eqw, Ehqw, EhRqw
+# Deleted     all running and suspended states with deletion     dr, dt, dRr, dRt,
+#                                                                ds, dS, dT, dRs,
+#                                                                dRS, dRT
+
+
+class SGEState(QSubState):
+    # Queue states
```

Comment:
The point is the the `SGEState` should contain the state of the submitted jobs, not of the queue. Maybe we can leave them for safety, in case they show up also as states of the Job? But if we are sure that they are queue states I think they can be removed.

Reply:
I guess I can remove the queue states and we see hat happens during testing if we somehow still need them.